examples/hwapi/README.md

```diff
@@ -0,0 +1,126 @@
+This directory shows the best practices for using MicroPython hardware API
+(`machine` module). `machine` module strives to provide consistent API
+across various boards, with the aim to enable writing portable applications,
+which would work from a board to board, from a system to another systems.
+This is inherently a hard problem, because hardware is different from one
+board type to another, and even from examplar of board to another. For
+example, if your app requires an external LED, one user may connect it
+to one GPIO pin, while another user may find it much more convinient to
+use another pin. This of course applies to relays, buzzers, sensors, etc.
+
+With complications above in mind, it's still possible to write portable
+applications by using "low[est] denominator" subset of hardware API and
+following simple rules outlines below. The applications won't be able
+to rely on advanced hardware capabilities of a particular boards and
+will be limited to generic capabilities, but it's still possible to
+write many useful applications in such way, with the obvious benefit of
```

Comment:
Hmm, I even googled that up: https://www.usingenglish.com/forum/threads/2521-quot-in-such-way-quot-vs-quot-in-such-a-way-quot . Ok, "in such way" is colloquial.